app/editor/src/features/content/form/ContentClipForm.tsx

```diff
@@ -59,14 +74,27 @@ export const ContentClipForm: React.FC<IContentClipFormProps> = ({
   };
 
   const onDownload = async (item: IItemModel) => {
-    await storage.download(`${folder.path}/${item.name}`);
+    await storageApi.download(`${folder.path}/${item.name}`);
   };
 
   const onAttach = async (item: IItemModel) => {
-    await storage.attach(content.id, `${folder.path}/${item.name}`).then((data) => {
-      setContent(toForm(data));
-      toast.success('Attachment added to this snippet.');
-    });
+    if (values.id === 0) {
+      // Add a reference to the file so that it can be copied to the API when the content is saved.
+      setFieldValue('fileReferences', [
```

Comment:
Here's the magic to attach the file to the content.  However, this isn't the complete picture because this file will not actually be accessible to the API endpoints until the file is copied over.